### DISCUSSION TITLE
能否动态注册schemaField

### DISCUSSION THREAD

**Discussion by Jiazhi-Wong** [alibaba/formily#3184] [Q&A 2.x] [answered]
如图，这是注册表单组件schemaField的方式。

![image](https://user-images.githubusercontent.com/19643491/173305361-cbd46a09-01c8-46f2-b503-055b4986d453.png)

我们业务有个场景需要动态注册，先注册基础组件，通过接口获取到异步数据后，再动态注册业务组件，有相关api支持吗？文档中没找到

**Comment by janryWang** [comment, marked as answer]
SchemaField组件可以传components和scope
